specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -99,23 +120,25 @@ export function Notifications({
                   type,
                   payload: rest as IR<LocalizedString>,
                 })
+              ),
+            ].sort(
+              sortFunction(
+                ({ timestamp }) => new Date(timestamp).getTime(),
+                true
               )
-              // Make most recent notification first
-              .sort(
-                sortFunction(
-                  ({ timestamp }) => new Date(timestamp).getTime(),
-                  true
-                )
-              )
-          );
-          // Stop updating if tab is hidden
-          timeout =
-            document.visibilityState === 'hidden'
-              ? undefined
-              : globalThis.setTimeout(() => doFetch(new Date()), pullInterval);
-          return undefined;
+            );
+          });
         })
         .catch(console.error);
+
+      // Stop updating if tab is hidden
+      timeout =
```

Comment:
I don't think this should be outside the `.then()` or else you risk starting fetching notifications again even before previous notifications have finished fetching

Reply:
https://www.techtarget.com/searchstorage/definition/race-condition?Offer=abMeterCharCount_var1